### PR TITLE
Add geofence with google play service example

### PR DESCRIPTION
**Pull request by manhinhang**
Added example showing how to toggling location service base on Geofencing with Google play service

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,7 +1,13 @@
 
-# indoorAtlasApiKey=api-key-here
-# indoorAtlasApiSecret=api-secret-here
+#indoorAtlasApiKey=8d000b08-1048-4894-b540-3e9f670b94e3
+#indoorAtlasApiSecret=njjAl75J1ygYv79XCOVBgykeklK9kReX0Zroyrw/wnZ0b9v/r6HgBYXLoMCchrDKvDOyiXaeAjd17QzA0qPgAxKPq0fYbfh34RIMhr2baMp+75SRr0dw9IqXYV66Jw==
+
+
+indoorAtlasApiKey=a55c7e6e1-6b67-4ba4-bbef-afadbc10a68b
+indoorAtlasApiSecret=Q6MfIEcPr+zHorcGuIU+lIxWjhOJfMnQqwKTIzhpZGNLU9Pg/jSbncr9dGFaaA7JXivd2YDplgo1QbNOdlJRil2iJOXzibs+BmDusMQjH85C0tTubE90ARQEk3IjaQ==
```
Comment (Cloudef) [inline gradle.properties:7]: Probably do not want these here 😓

Comment (manhinhang) [reply gradle.properties]: Sorry for that  🙇‍♂️ removed now

**Review by Cloudef — APPROVED** [review]

File: Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    public static boolean checkBackgroundLocationPermissions(Activity activity) {
+        return ContextCompat.checkSelfPermission(activity, Manifest.permission.ACCESS_BACKGROUND_LOCATION)
+                == PackageManager.PERMISSION_GRANTED;
+    }
+
+    private void ensurePermissions() {
+
+        if (!checkBackgroundLocationPermissions(this)) {
+            // We don't have access to FINE_LOCATION (Required by Google Maps example)
+            // IndoorAtlas SDK has minimum requirement of COARSE_LOCATION to enable WiFi scanning
```
Comment (matti-ida) [inline Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java:170]: This comment does not seem to match code.

File: Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+@SdkExample(description = R.string.example_system_geofencing_description)
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+
+    private final static int REQUEST_CODE_ACCESS_COARSE_LOCATION = 1;
```
Comment (matti-ida) [inline Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java:48]: I think this is also a bit misleading name, has nothing to do with coarse location (it's just arbitrary "magic number").

File: Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        loadingBar = findViewById(R.id.progressBar);
+        defaultGeofenceCoordText = findViewById(R.id.textView1);
+        registerGeofenceButton.setOnClickListener(this);
+        unregisterGoefenceButton.setOnClickListener(this);
+
+        geofencingClient = LocationServices.getGeofencingClient(this);
+        ensurePermissions();
+
+        // Loading venue center coordinate from Positioning API
+        String url = String.format("https://positioning-api.indooratlas.com/v1/venues?key=%s", getString(R.string.indooratlas_api_key));
```
Comment (matti-ida) [inline Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java:76]: Maybe somewhere should be noted the api key must have "positioning API" scope.

Comment (matti-ida) [reply Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java:76]: (in addition to SDK scope)

File: Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java (hunk 2 above)
Comment (manhinhang) [reply Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java]: Fixed typo and updated description why required background location permission

File: Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java (hunk 4 above)
Comment (manhinhang) [reply Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java:76]: The "positioning API" scope is optional. If the api key can not retrieve the coordinate, it use hardcoded LatLng by default and UI show "Unable to load venue coordinate from Positioning API. **Use default coordinate defined in source code.**" message to notice developer

Comment (matti-ida) [reply Basic/src/main/java/com/indooratlas/android/sdk/examples/systemgeofence/MainActivity.java:76]: Right, could still add comment though :)

**Review by matti-ida — APPROVED** [review]

**Review by nelsoncheung-indooratlas — APPROVED** [review]

**Pull request merged**